refactor: remove lodash from runtime dependencies

M-TGH opened the pull request:
*Issue #, if available:* #84

*Description of changes:*

Removes use of lodash functions making lodash no longer required on runtime.
May have a slight regression in performance on the replacement of omit. 
Would prefer using destructuring but that's not supported below node v6.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

chrisradek commented on packages/core/lib/segments/attributes/subsegment.js:
```diff
@@ -124,15 +116,15 @@ Subsegment.prototype.addPrecursorId = function(id) {
 
 Subsegment.prototype.addAnnotation = function(key, value) {
-  if (!(isBoolean(value) || isString(value) || isFinite(value))) {
+  if (!['boolean', 'number', 'string'].includes(typeof value)) {
```
I like how you've kept the conditional compact, but if the type is a number we also want to make sure it is finite (not Infinity or NaN) using the native `isFinite` function. Something like this:

```suggestion
  if (!(typeof value === 'boolean' || typeof value === 'string' || (typeof value === 'number' && isFinite(value))) {
```

chrisradek commented on packages/core/lib/segments/attributes/subsegment.js:
```diff
@@ -379,12 +371,16 @@ Subsegment.prototype.toString = function toString() {
-  if (isEmpty(this.subsegments))
-    ignore.push('subsegments');
+  if (Array.isArray(this.subsegments) && !this.subsegments.length && 'subsegments' in thisCopy)
+    delete thisCopy.subsegments;
```
Nit: Since the body of the if statement is on a separate line, can you wrap this in brackets? That'll make it more clear what our intent is.

chrisradek commented on packages/core/lib/segments/segment.js:
```diff
@@ -346,18 +338,26 @@ Segment.prototype.close = function(err, remote) {
-  if (isEmpty(this.subsegments))
-    ignore.push('subsegments');
+  if (Array.isArray(this.subsegments) && !this.subsegments.length && 'subsegments' in thisCopy)
+    delete thisCopy.subsegments;
```
Nit: Same as above. Since the body of the if statement is on a separate line, can you wrap this in brackets? That'll make it more clear what our intent is.

M-TGH commented:
Thanks for the feedback, I've added the brackets to the long if statements in 194a023.
Added isFinite in 2f11c21 but was missing a closing bracket, fixed in 0816dd0.

chrisradek commented:
@M-TGH 
Thanks for making the changes and for your contribution! Merging!
